src/objects.h

```diff
@@ -614,10 +615,18 @@ extern void CheckedMakeImmutable( Obj obj );
 **  'IS_MUTABLE_OBJ' returns   1 if the object  <obj> is mutable   (i.e., can
 **  change due to assignments), and 0 otherwise.
 */
-#define IS_MUTABLE_OBJ(obj) \
-                        ((*IsMutableObjFuncs[ TNUM_OBJ(obj) ])( obj ))
-
 extern Int (*IsMutableObjFuncs[LAST_REAL_TNUM+1]) ( Obj obj );
+static inline Int IS_MUTABLE_OBJ(Obj obj)
+{
+    if(IS_INTOBJ(obj) || IS_FFE(obj)) {
+        return 0;
+    } else if (TEST_OBJ_FLAG(obj, OBJ_FLAG_IMMUTABLE) == OBJ_FLAG_IMMUTABLE) {
```

Comment:
As they are set up OBJ_FLAGS are universal -- the meaning or value of a flag is not bound to specific TNUMs for which it makes sense. 

So it would seem logical to make sure each flag is set appropriately even for those TNUMs where there is only one appropriate setting. So all permutations would have the immutable flag, for instance (or not have the mutable flag).   Unless this creates some run-time overhead I haven't thought of, then it seems simpler. The only problem would be if we ran out of flag bits.  

That leaves T_INTOBJ and T_FFE. I think the nicest way to handle those (and the most consistent with what is done for those objects elsewhere) is to distinguish BAG_FLAGS from OBJ_FLAGS and deal with the two immediate types in the OBJ_FLAGS code. So Gasman never has to worry about those types and higher level code sees flags consistently on every object.